metrics/cpu_enabled.go

```diff
@@ -14,7 +14,7 @@
 // You should have received a copy of the GNU Lesser General Public License
 // along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
 
-// +build !ios
+// +build !ios,!darwin
```

Comment:
I think you should be able to cover all cases here and not need that extra file:

```go
// +build !ios
// +build !darwin cgo
```

Could you try this version out?

Reply:
I think it should be
```suggestion
// +build !ios,!darwin darwin,cgo
```

Reply:
This is the truth table I'm using

ios | darwin | Other OS | cgo | cpu enabled
-- | -- | -- | -- | --
0 | 1 | 0 | 0 | 0
0 | 1 | 0 | 1 | 1
1 | 0 | 0 | 0 | 0
1 | 0 | 0 | 1 | 0
0 | 0 | 1 | 0 | 1
0 | 0 | 1 | 1 | 1

Reply:
So the **Other OS** will essentially be caught by `!ios,!darwin`, correct? If this is the case, the change you suggested makes sense.